platform/logging/LoggingConfig.ts

```diff
@@ -0,0 +1,180 @@
+import { Schema, typeOfSchema } from '../types';
+import { LogLevelId } from './LogLevel';
+import { Appenders, AppenderConfigType } from './appenders/Appenders';
+
+/**
+ * Separator string that used within nested context name (eg. plugins::pid).
+ */
+const CONTEXT_SEPARATOR = '::';
+
+/**
+ * Name of the `root` context that should always be defined.
+ */
+const ROOT_CONTEXT_NAME = 'root';
+
+const defaultAppenders = new Map([
+  [
+    'console',
+    {
+      kind: 'console',
+      layout: {
+        kind: 'pattern',
+        highlight: true
+      }
+    }
+  ]
+]);
+
+const defaultLoggers = new Map([
+  ['root', { appenders: ['console'], level: 'info' as LogLevelId }]
+]);
+
+const createLoggerSchema = (schema: Schema) => {
+  const { literal, object, string, arrayOf, oneOf } = schema;
+
+  return object({
+    level: oneOf(
+      [
+        literal('all'),
+        literal('fatal'),
+        literal('error'),
+        literal('warn'),
+        literal('info'),
+        literal('debug'),
+        literal('trace'),
+        literal('off')
+      ],
+      {
+        defaultValue: 'info'
+      }
+    ),
+    appenders: arrayOf(string(), { defaultValue: [] })
+  });
+};
+
+const createLoggingSchema = (schema: Schema) => {
+  const { object, string, mapOf } = schema;
+
+  return object({
+    appenders: mapOf(string(), Appenders.createConfigSchema(schema), {
+      defaultValue: defaultAppenders
+    }),
+    loggers: mapOf(string(), createLoggerSchema(schema), {
+      defaultValue: defaultLoggers,
+      validate(value) {
+        const rootLogger = value.get(ROOT_CONTEXT_NAME);
+        if (!rootLogger) {
+          throw new Error(
+            `"${ROOT_CONTEXT_NAME}" logger should be configured!`
```

Comment:
Yes, that is what I was complaining about here https://github.com/elastic/kibana/pull/12852#discussion_r128997798.

It definitely has to be improved, but I'd rather experiment with it a dedicated follow-up. Does it work for you?